gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactor.java

```diff
@@ -65,8 +99,14 @@
   private final String destSubDir;
   private final String tmpDir;
   private final FileSystem fs;
-  private final ExecutorService executorService;
-  private final List<Future<?>> futures;
+  private final Map<String, State> inputFolderJobProps;
```

Comment:
Please define a record type instead of having three different maps with the same key.